src/Bicep.Core/Syntax/MalformedIdentifierSyntax.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parser;
+
+namespace Bicep.Core.Syntax
+{
+    /// <summary>
+    /// Represents a malformed or missing identifier.
+    /// </summary>
+    public class MalformedIdentifierSyntax : IdentifierSyntaxBase
+    {
+        public MalformedIdentifierSyntax(IEnumerable<Token> tokens, IEnumerable<Diagnostic> diagnostics)
+        {
+            this.Tokens = tokens.ToImmutableArray();
+            if (this.Tokens.Any() == false)
```

Comment:
Out of interest, why do `(x == false)` on a non-nullable method call? Personally when reading, I find it much quicker to grok `!x` as "not x" - for some reason my brain has to do a conversion on `x == false` -> `x != true` -> `!x`.

Reply:
Just a style thing - easy to miss `!`.